lib/ansible/modules/packaging/os/flatpak.py

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+DOCUMENTATION = '''
+---
+module: flatpak
+version_added: "2.4"
+requirements:
+    - flatpak
```

Comment:
The flatpak command is listed as a requirement but the module doesn't check if it is before proceeding. It would be a good idea to `module.fail_json()` with a useful message when that occurs. You can actually even get the full path with the function `module.get_bin_path()` as illustrated here: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/system/service.py#L451
That would also get you a full path to execute so that you don't run into relative path issues.